### PR TITLE
Merge lists using a for-loop instead of unionBy

### LINKED ISSUE #778: Subscription being reloaded unnecessarily
**Describe the bug**

Found this while testing a PR with @hjoliver https://github.com/cylc/cylc-ui/pull/749#issuecomment-923317099

Copied the `initialQuery` and `finalQuery`, and realized it's a bug caused by https://github.com/cylc/cylc-ui/pull/726

The diff shows that some values were incorrectly merged.

**Release version(s) and/or repository branch(es) affected?**

`master`

**Steps to reproduce the bug**

1. Add a Tree view
2. Add a second Tree view
3. Either observe in Network tab or see if the components are cleared, leaving a blank view, then reloaded

**Expected behavior**

Since both views use the same query, nothing must change in the UI, except for a new view added.

**Screenshots**

**Additional context**

```diff
kinow@ranma:/tmp$ diff initial final
2c2
<   deltas(workflows: [$workflowId], stripNull: true) {
---
>   deltas(workflows: [$workflowId, $workflowId], stripNull: true) {
24c24
<   cyclePoints: familyProxies(ids: ["root"], ghosts: true) {
---
>   cyclePoints: familyProxies(ids: ["root", "root"], ghosts: true) {
27c27
<   familyProxies(exids: ["root"], sort: {keys: ["name"]}, ghosts: true) {
---
>   familyProxies(exids: ["root", "root"], sort: {keys: ["name"]}, ghosts: true) {
45c45
<   familyProxies(exids: ["root"], ghosts: true) {
---
>   familyProxies(exids: ["root", "root"], ghosts: true) {
65c65
<     states: ["failed", "preparing", "submit-failed", "submitted", "running"]
---
>     states: ["failed", "preparing", "submit-failed", "submitted", "running", "failed", "preparing", "submit-failed", "submitted", "running"]

```

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

### PR DESCRIPTION
**Pull request by kinow**
These changes close #778

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] No change log entry required (why? e.g. invisible to users, not released yet).
- [x] No documentation update required.

### CODE REVIEW

File: tests/unit/graphql/merge.spec.js [hunk 1]
```diff
@@ -204,40 +204,60 @@ describe('mergeQueries', () => {
...
+            }
+          }`
+          const queryB = gql`query B {
+            jobs (workflows: ["root", "airplane"]) {
+              name
+            }
+          }`
+          const merged = mergeQueries(queryA, queryB)
+          const expected = gql`query A {
+            jobs (workflows: ["root", "test", "airplane"]) {
```
Comment (kinow) [inline tests/unit/graphql/merge.spec.js:254]: This is essentially what was happening when merging the subscriptions; see linked issue for the complete diff.

**Comment by kinow** [comment]
To review:

1. Go to `master`, build it for development
2. Open a tree view
3. Add a second tree view, watch either Network tab for the stop/start messages, or see if the UI clears the views and re-populates them

Then switch to this branch and try that again.

**Review by oliver-sanders — COMMENTED** [review]
Code looks good, haven't tested yet (my clones are in a mess).

**Review by oliver-sanders — APPROVED** [review]
Tested, the subscription wasn't stopped / restarted :+1:.

**Review by hjoliver — APPROVED** [review]

**Pull request merged**